src/services/storage.rs

```diff
@@ -35,14 +47,36 @@ impl fmt::Debug for StorageService {
 
 impl StorageService {
     /// Creates a new storage service instance with specified storage area.
-    pub fn new(area: Area) -> Self {
-        let storage = {
-            match area {
-                Area::Local => window().local_storage(),
-                Area::Session => window().session_storage(),
-            }
+    pub fn new(area: Area) -> Result<Self, &'static str> {
+        let storage = cfg_match! {
+            feature = "std_web" => ({
+                let storage_name = match area {
+                    Area::Local => "localStorage",
+                    Area::Session => "sessionStorage",
+                };
+                let storage = js! {
+                    try {
+                        return window[@{storage_name}];
+                    } catch(error) {
+                        return error;
+                    }
+                };
+                Storage::try_from(js!( return @{storage.as_ref()}; ))
```

Comment:
why did you change the stdweb implementation here?

Reply:
From our discussion above, we agreed that we want this function returns an error when the user is blocking storage.
`stdweb` isn't capable of that, it will always panic instead.
If I revert this, then the user can only handle this case on `web-sys`, but not on `stdweb`.

Reply:
Got it, sounds good! It'd be nice to upstream this to stdweb but I don't think that would be worth the effort due to the lack of github activity for that project.

Reply:
> It'd be nice to upstream this to stdweb but I don't think that would be worth the effort due to the lack of github activity for that project.

Agreed.